python/_tskitmodule.c

```diff
@@ -736,6 +736,32 @@ make_file(PyObject *fileobj, const char *mode)
     return ret;
 }
 
+static int
+uint32_converter(PyObject *py_obj, uint32_t *uint_out)
+{
+    int64_t temp_long;
```

Comment:
This needs to be declared a ``long long`` I think, as that's what the python C code is assuming. It could happen that the long long is more than 64 bits or something - anyway, no point in mixing fixed widths and old style types. We can be sure a long long is at least 64 bits.